bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java

```diff
@@ -304,12 +318,12 @@ public static void startCallableObservation(Environment env, BString module, BSt
         if (prevObserverCtx != null) {
             newObContext.setServiceName(prevObserverCtx.getServiceName());
             newObContext.setEntrypointFunctionModule(prevObserverCtx.getEntrypointFunctionModule());
-            newObContext.setEntrypointFunctionPosition(prevObserverCtx.getEntrypointFunctionPosition());
+            newObContext.setEntrypointFunctionName(prevObserverCtx.getEntrypointFunctionName());
```

Comment:
We need to set entrypoint function and object both here since we need both to identify a function call.

Reply:
Fixed in cc06a1391b7e2bbe4071119cfa474801cee21e57

Reply:
Fixed in deebe9e0fbd82b278332e377842f2c99023dae07